R/distribution.R

```diff
@@ -380,40 +380,42 @@ Distribution <-   function(x,
         density.color <- rep(density.color, n.variables)
     if (length(density.color) != n.variables)
         warning("The number of colors provided for shading the densities is not consistent with the number of variables.")
-
     # Histograms can be specified using 'maximum.bins' or 'size' (which overrides the former)
     # 'maximum.bins' is not always respected so use 'size' instead when
     # bin size is small or the number of bins is small
     # but otherwise use 'maximum.bins' because it tends to find more rounded breakpoints
     x.sorted <- sort(unique(unlist(x)))
     rng <- x.sorted[c(1, length(x.sorted))]
-    default.bins <- is.null(maximum.bins) || is.na(maximum.bins)
-    if (is.null(maximum.bins) || is.na(maximum.bins))
-        maximum.bins <- min(length(x.sorted), 50)
-    bin.min.size <- min(diff(x.sorted))
     if (density.type == "Histogram")
     {
+        bin.min.size <- min(diff(x.sorted))
```

Comment:
I wonder if we should have done something different if the values are non-integer, to avoid tiny bin sizes.

Reply:
`bin.min.size` is not actually used unless the sample has only a few unique values. But we use it to identify when plotly's default might run into problems. Added a comment in [d1ef499](https://github.com/Displayr/flipStandardCharts/pull/64/commits/d1ef4995aa959850235be7795b887f7942a85de2) to explain this

Reply:
Don't we use `bin.min.size` to compute `maximum.bins` on line 402? If `bin.min.size` is small then `maximum.bins` could be large.